vpp-manager/uplink/rdma.go

```diff
@@ -0,0 +1,139 @@
+// Copyright (C) 2020 Cisco Systems Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package uplink
+
+import (
+	"fmt"
+
+	"github.com/pkg/errors"
+	"github.com/projectcalico/vpp-dataplane/vpp-manager/config"
+	"github.com/projectcalico/vpp-dataplane/vpp-manager/utils"
+	"github.com/projectcalico/vpp-dataplane/vpplink"
+	log "github.com/sirupsen/logrus"
+)
+
+type RDMADriver struct {
+	UplinkDriverData
+}
+
+func (d *RDMADriver) IsSupported(warn bool) (supported bool) {
+	var ret bool
+	supported = true
+	ret = d.params.LoadedDrivers[config.DRIVER_VFIO_PCI]
+	if !ret && warn {
+		log.Warnf("did not find vfio-pci or uio_pci_generic driver")
+		log.Warnf("VPP may fail to grab its interface")
+	}
+	supported = supported && ret
+
+	ret = d.params.AvailableHugePages > 0
+	if !ret && warn {
+		log.Warnf("No hugepages configured, driver won't work")
+	}
+	supported = supported && ret
+
+	ret = d.conf.Driver == config.DRIVER_VIRTIO_PCI
+	if !ret && warn {
+		log.Warnf("Interface driver is <%s>, not %s", d.conf.Driver, config.DRIVER_VIRTIO_PCI)
+	}
+	supported = supported && ret
+
+	return supported
+}
+
+func (d *RDMADriver) PreconfigureLinux() (err error) {
+	newDriverName := d.params.NewDriverName
+	doSwapDriver := d.conf.DoSwapDriver
+	/*if newDriverName == "" {
```

Comment:
Please remove everything related to driver swap since it is not required for this driver.